### PR TITLE
Support readonly resource type generation

### PR DESCRIPTION
**Pull request by jason-dou**
Support generating readonly Graph Bicep resources
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoftgraph/msgraph-bicep-types/pull/186)

### CODE REVIEW

File: src/swagger-generation/output/microsoftgraph-beta-0.1.8-preview.json [hunk 1]
```diff
@@ -1429,7 +1439,7 @@
           "groups"
         ],
         "description": "Create or update a group",
-        "operationId": "groups_Put",
+        "operationId": "groups_put",
```
Comment (dkershaw10) [inline src/swagger-generation/output/microsoftgraph-beta-0.1.8-preview.json:1442]: Is it really a put or an upsert?

File: src/swagger-generation/output/microsoftgraph-beta-0.1.8-preview.json [hunk 2]
```diff
@@ -1456,7 +1466,7 @@
         ],
         "responses": {
           "200": {
-            "description": "group created/updated successfully",
+            "description": "group create or update successfully",
```
Comment (dkershaw10) [inline src/swagger-generation/output/microsoftgraph-beta-0.1.8-preview.json:1469]:
```suggestion
            "description": "Group created or updated successfully",
```
Same in other places

**Review by dkershaw10 — APPROVED** [review]
Approved with a couple of comments.
Nice to see this go in to support the User entity type.

File: src/swagger-generation/output/microsoftgraph-beta-0.1.8-preview.json (hunk 1 above)
Comment (jason-dou) [reply src/swagger-generation/output/microsoftgraph-beta-0.1.8-preview.json:1442]: The operation http method is `put`. Name doesn't really matter though I can change it to upsert

**Review by dkershaw10 — APPROVED** [review]
LGTM

**Pull request merged**